src/context/WalletProvider/WalletViewsSwitch.tsx

```diff
@@ -88,7 +89,8 @@ export const WalletViewsSwitch = () => {
     history.goBack()
     // If we're back at the select wallet modal, remove the initial route
     // otherwise clicking the button for the same wallet doesn't do anything
-    if (history.location.pathname === INITIAL_WALLET_MODAL_ROUTE) {
+    const { pathname } = history.location
+    if ([INITIAL_WALLET_MODAL_ROUTE, NativeWalletRoutes.Load].includes(pathname)) {
```

Comment:
Specifically, add `NativeWalletRoutes.Load` is the fix.

This whole routing pattern is a bit of a mess, but I don't want to refactor it all in this fix.

Reply:
Yeah this is very close from what I was planning to push today, it's a mess but the refactoring would be a bit bigger